### PR TITLE
Fallback to config in external strategy

### PR DESCRIPTION
**Pull request by jmguzik**
Fallback to cluster configured in the config file when unsuccessfull with scheduling for external strategy

### CODE REVIEW

File: pkg/scheduler/strategy/external.go [hunk 1]
```diff
@@ -99,7 +102,8 @@ func (e *External) Schedule(_ context.Context, pj *prowv1.ProwJob) (Result, erro
 
 	resp, err := query(e.cfg.URL, SchedulingRequest{Job: pj.Spec.Job})
 	if err != nil {
-		return Result{}, fmt.Errorf("error querying URL: %w", err)
+		logrus.WithField("job", pj.Spec.Job).WithField("cluster", pj.Spec.Cluster).Warn("scheduling failed, using Spec.Cluster entry")
```
Comment (smg247) [inline pkg/scheduler/strategy/external.go:105]: you aren't using the logger that you added to `External`

File: pkg/scheduler/strategy/__debug_bin1472196796
Comment (smg247) [inline pkg/scheduler/strategy/__debug_bin1472196796:1]: I assume this is an accident?

Comment (jmguzik) [reply pkg/scheduler/strategy/__debug_bin1472196796:1]: yes :(

File: pkg/scheduler/strategy/external_test.go [hunk 2]
```diff
@@ -124,6 +125,7 @@ func TestExternalSchedule(t *testing.T) {
 				},
 				cache:     tt.cache,
 				timestamp: time.Now().Add(-3 * time.Hour), // To trigger cache cleanup if mockCleanup is true
+				log:       logrus.NewEntry(logrus.New()),
```
Comment (droslean) [inline pkg/scheduler/strategy/external_test.go:128]:
```suggestion
				log:       logrus.WithField("name", tt.name)
```

**Pull request merged**